### PR TITLE
feat(dli): add new datasource to get list of flink templates

### PR DESCRIPTION
**Pull request by wangyuancheng01**
…atasource to get list of flink templates

**What this PR does / why we need it**:
   add new datasource to get list of flink templates.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/dli" TESTARGS="-run TestAccDataSourceDliFlinkTemplates_basic"
...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDataSourceDliFlinkTemplates_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDliFlinkTemplates_basic
=== PAUSE TestAccDataSourceDliFlinkTemplates_basic
=== CONT  TestAccDataSourceDliFlinkTemplates_basic
--- PASS: TestAccDataSourceDliFlinkTemplates_basic (40.39s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       40.479s
```

### CODE REVIEW

File: huaweicloud/services/dli/data_source_huaweicloud_dli_flink_templates.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+}
+
+func dataSourceDliFlinkTemplatesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	wrapper := newFlinkTemplatesDSWrapper(d, meta)
+	lisFliSqlJobTemRst, err := wrapper.ListFlinkSqlJobTemplates()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, _ := uuid.GenerateUUID()
```
Comment (deer-hang) [inline huaweicloud/services/dli/data_source_huaweicloud_dli_flink_templates.go]: The error information returned by the function cannot be omitted.

File: huaweicloud/services/dli/data_source_huaweicloud_dli_flink_templates.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+					"created_at":  w.setTemLisTemCreTim(body, &tem),
+					"updated_at":  w.setTemLisTemUpdTim(body, &tem),
+				}
+			},
+		)),
+	)
+	return mErr.ErrorOrNil()
+}
+
+func (*FlinkTemplatesDSWrapper) setTemLisTemTemId(_, data *gjson.Result) string {
```
Comment (deer-hang) [inline huaweicloud/services/dli/data_source_huaweicloud_dli_flink_templates.go]: The first parameter is not applied.

File: huaweicloud/services/dli/data_source_huaweicloud_dli_flink_templates.go (hunk 1 above)
Comment (wangyuancheng01) [reply huaweicloud/services/dli/data_source_huaweicloud_dli_flink_templates.go]: done

File: huaweicloud/services/dli/data_source_huaweicloud_dli_flink_templates.go (hunk 2 above)
Comment (wangyuancheng01) [reply huaweicloud/services/dli/data_source_huaweicloud_dli_flink_templates.go]: done

**Pull request merged**